learn2track/models/gru_regression.py

```diff
@@ -219,9 +210,11 @@ def _compute_losses(self, model_output):
         if self.normalize_output:
             regression_outputs /= (T.sqrt(T.sum(regression_outputs**2, axis=2, keepdims=True) + self.eps))
 
-        # L2_errors_per_time_step.shape = (batch_size,)
-        self.L2_errors_per_time_step = T.sqrt(T.sum(((regression_outputs - self.dataset.symb_targets)**2), axis=2))
-        # avg_L2_error_per_seq.shape = (batch_size,)
-        self.avg_L2_error_per_seq = T.sum(self.L2_errors_per_time_step*mask, axis=1) / T.sum(mask, axis=1)
+        self.samples = regression_outputs
+
+        # loss_per_time_step.shape = (batch_size,)
+        self.loss_per_time_step = T.sqrt(T.sum(((self.samples - self.dataset.symb_targets)**2), axis=2))
```

Comment:
This line and the one below are being used a lot (different loss implementation for different models). It can wait, but it would be easy to refactored it out in a `losses_utils.py`.